core/actor-zeta/flat_map.hpp

```diff
@@ -0,0 +1,235 @@
+  using const_pointer          = typename vector_type::const_pointer;
+  using iterator               = typename vector_type::iterator;
+  using const_iterator         = typename vector_type::const_iterator;
+  using reverse_iterator       = typename vector_type::reverse_iterator;
+  using const_reverse_iterator = typename vector_type::const_reverse_iterator;
+
+  flat_map() noexcept = default;
+
+  ~flat_map() noexcept = default;
+
+  flat_map(const vector_type& r) noexcept
+    : cache_(r)
+  { }
+
+  flat_map(std::initializer_list<value_type> l) noexcept
+    : cache_(l)
+  { }
+
+  template <class InputIterator>
+  flat_map(InputIterator first, InputIterator last) {
+    cache_.insert(cache_.end(), first, last);
+  }
+
+  iterator begin() noexcept {
+    return cache_.begin();
+  }
+
+  const_iterator begin() const noexcept {
+    return cache_.begin();
+  }
+
+  const_iterator cbegin() const noexcept {
+    return cache_.cbegin();
+  }
+
+  reverse_iterator rbegin() noexcept {
+    return cache_.rbegin();
+  }
+
+  const_reverse_iterator rbegin() const noexcept {
+    return cache_.rbegin();
+  }
+
+  iterator end() noexcept {
+    return cache_.end();
+  }
+
+  const_iterator end() const noexcept {
+    return cache_.end();
+  }
+
+  const_iterator cend() const noexcept {
+    return cache_.cend();
+  }
+
+  reverse_iterator rend() noexcept {
+    return cache_.rend();
+  }
+
+  const_reverse_iterator rend() const noexcept {
+    return cache_.rend();
+  }
+
+  bool empty() const noexcept {
+    return cache_.empty();
+  }
+
+  size_type size() const noexcept {
+    return cache_.size();
+  }
+
+  void reserve(size_type count) {
+    cache_.reserve(count);
+  }
+
+  void shrink_to_fit() {
+    cache_.shrink_to_fit();
+  }
+
+  vector_type& container() noexcept {
+    return cache_;
+  }
+
+  const vector_type& container() const noexcept {
+    return cache_;
+  }
+
+  void clear() noexcept {
+    cache_.clear();
+  }
+
+  void swap(flat_map& other) {
+    cache_.swap( other.cache_ );
+  }
+
+  std::pair<iterator, bool> insert(value_type x) {
+    for(auto it = cache_.begin(); it != cache_.end(); ++it)
+      if(*it == x)
+        return std::pair<iterator, bool>( it, false ); //already exists
+    return std::pair<iterator, bool>( cache_.insert(cache_.end(), x), true ); //inserted
+  }
+
+  iterator insert(iterator hint, value_type x) {
+    return cache_.insert(cache_.end(), x); //should be optimized with the hint
+  }
+
+  iterator insert(const_iterator hint, value_type x) {
+    return cache_.insert(cache_.end(), x); //should be optimized with the hint
+  }
+
+  template <class InputIterator>
+  void insert(InputIterator first, InputIterator last) {
+    for(auto it = first; it != last; ++it)
+      cache_.insert( cache_.end(), *it );
+  }
+
+  template <class... Ts>
+  std::pair<iterator, bool> emplace(Ts&&... xs) {
+    const iterator &emplaced = cache_.emplace(cache_.end(), std::forward<Ts>(xs)...);
```

Comment:
Why not using a method insert ?

Reply:
I guess it's faster than insert/push_back. Is not is?